java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java

```diff
@@ -640,4 +641,47 @@ CompletableFuture<Map<String, Double>> zrangeWithScores(
      * }</pre>
      */
     CompletableFuture<Long> zremrangebyscore(String key, ScoreRange minScore, ScoreRange maxScore);
+
+    /**
+     * Computes the intersection of sorted sets given by the specified <code>keys</code>, and stores
+     * the result in <code>destination</code>. If <code>destination</code> already exists, it is
+     * overwritten. Otherwise, a new sorted set will be created.
+     *
+     * @see <a href="https://redis.io/commands/zinterstore/">redis.io</a> for more details.
+     * @param destination The key of the destination sorted set.
+     * @param keys The keys of sorted sets.
+     * @param options Weight and Aggregate options.
+     * @return The number of elements in the resulting sorted set.
+     * @example
+     *     <pre>{@code
+     * WeightAggregateOptions options =
+     *     WeightAggregateOptions.builder()
+     *             .aggregate(Aggregate.MAX)
+     *             .weights(List.of(1.0, 2.0))
```

Comment:
maybe your example should use multiple .weight(double)? Since you're adding the @singular tag, we should use it?

Reply:
Why would someone not want to pass a list? (Seems easier to do)